pkg/my/domain.go

```diff
@@ -13,45 +16,51 @@ type Domain struct {
 	Address string `json:"address"`
 }
 
-type Domains struct {
-	file    string
-	domains []Domain
-}
-
-func (d *Domains) List() []Domain {
-	return d.domains
-}
-
-func (d *Domains) Save() error {
-	cnt, err := json.Marshal(d.domains)
+func Domains(client *node.Client, nickname string) ([]string, error) {
+	const ownedPrefix = "owned"
+	wallet, err := wallet.Load(nickname)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	err = os.WriteFile(d.file, cnt, 0o600) // u+r, u+w
+	domains := []string{}
+	domainsEntry, err := node.DatastoreEntry(client, dns.DnsRawAddress, ownedPrefix+wallet.Address)
 	if err != nil {
-		return err
+		return nil, err
 	}
-
-	return nil
+	err = json.Unmarshal(domainsEntry.CandidateValue, &domains)
+	if err != nil {
+		return nil, err
+	}
+	return domains, nil
 }
 
-func (d *Domains) Add(dom Domain) {
-	d.domains = append(d.domains, dom)
-}
+func Websites(client *node.Client, domainNames []string) ([]*models.Websites, error) {
+	const recordPrefix = "record"
+
+	params := []node.GetDatastoreEntriesString{}
+	for i := 0; i < len(domainNames); i++ {
+		param := node.GetDatastoreEntriesString{
+			Address: dns.DnsRawAddress,
+			Key:     recordPrefix + domainNames[i],
+		}
+		params = append(params, param)
 
-func NewDomains() (*Domains, error) {
-	file, err := ioutil.ReadFile(domainFile)
-	if err != nil {
-		return nil, err
 	}
 
-	dom := Domains{file: domainFile, domains: []Domain{}}
+	responses := []*models.Websites{}
+	contractAddresses, err := node.DatastoreEntries(client, params)
 
-	err = json.Unmarshal(file, &dom.domains)
+	contractAddressess := *contractAddresses
```

Comment:
Try to avoid using additional variables that don't help understand the code.

If you solve the comment in getdatastoreentries, this variable will naturally disaper. But for your information you could have directly done : 
```go
	for i := 0; i < len(domainNames); i++ {
		response := models.Websites{
			Address: string((*contractAddresses)[i].CandidateValue),
			Name:    domainNames[i],
		}
		responses = append(responses, &response)
	}
```